thing/index.js

```diff
@@ -526,6 +572,8 @@ function ThingShadowsClient(deviceOptions, thingShadowOptions) {
 
    this.unregister = function(thingName) {
       if (thingShadows.hasOwnProperty(thingName)) {
+         var topicsObject = [];
```

Comment:
Use topicObjects to be consistent with your prior definition; and perhaps topicSpecs or toipcDefinitions is more readable?